pilot/pkg/networking/util/util.go

```diff
@@ -803,27 +804,44 @@ func BuildStatefulSessionFilter(svc *model.Service) *hcm.HttpFilter {
 }
 
 func MaybeBuildStatefulSessionFilterConfig(svc *model.Service) *statefulsession.StatefulSession {
-	if features.PersistentSessionLabel == "" || svc == nil {
+	if svc == nil {
 		return nil
 	}
-	sessionCookie := svc.Attributes.Labels[features.PersistentSessionLabel]
-	if sessionCookie == "" {
-		return nil
-	}
-	cookieName, cookiePath, found := strings.Cut(sessionCookie, ":")
-	if !found {
-		cookiePath = "/"
-	}
-	return &statefulsession.StatefulSession{
-		SessionState: &core.TypedExtensionConfig{
-			Name: "envoy.http.stateful_session.cookie",
-			TypedConfig: protoconv.MessageToAny(&cookiev3.CookieBasedSessionState{
-				Cookie: &httpv3.Cookie{
-					Path: cookiePath,
-					Ttl:  &durationpb.Duration{Seconds: 120},
-					Name: cookieName,
-				},
-			}),
-		},
+	switch {
+	case features.PersistentSessionLabel != "":
+		sessionCookie := svc.Attributes.Labels[features.PersistentSessionLabel]
```

Comment:
I think this is not expected. The expected res is when session label specified, you should set up session cookie. When session header specified, you should set up `session header` filter.  

However, with current implement, we always end up with  `envoy.http.stateful_session.cookie` set up

Reply:
I thought I fixed this. Sorry. realized I did not push it. PTAL now

Reply:
LG now